sdk_version_resolver.go

```diff
@@ -0,0 +1,156 @@
+package dotnetcoresdk
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Masterminds/semver"
+	"github.com/paketo-buildpacks/packit"
+	"github.com/paketo-buildpacks/packit/postal"
+)
+
+type RuntimeToSdks struct {
+	RuntimeVersion string   `toml:"runtime-version"`
+	SDKs           []string `toml:"sdks"`
+}
+
+type SDKVersionResolver struct {
+	logger LogEmitter
+}
+
+func NewSDKVersionResolver(logger LogEmitter) SDKVersionResolver {
+	return SDKVersionResolver{logger: logger}
+}
+
+func (r SDKVersionResolver) Resolve(path string,
```

Comment:
I understand the versioning logic outlined in the issue, but it's hard to decode what's going on in each step. Do you think you could add some comments to the code to make the logic a bit easier to follow?